clang/lib/Basic/Targets/AArch64.cpp

```diff
@@ -1059,7 +1059,7 @@ bool AArch64TargetInfo::handleTargetFeatures(std::vector<std::string> &Features,
       HasFullFP16 = true;
       HasFP16FML = true;
     }
-    if (Feature == "+mte")
+    if (Feature == "+mte2")
```

Comment:
This breaks backwards compatibility of IR. _If_ you want to go ahead with this patch (which is fundamentally just a change to the internal representation of features) then I would suggest breaking backwards compatibility by having `target-feature="+mte"` mean FEAT_MTE and `target-feature="+mte2"` mean FEAT_MTE2. Then if anyone happens to be using MTE2 instructions in inline assembly with `target-feature="+mte"` they will get an error, but for the vast majority of cases, the change from FEAT_MTE2 to FEAT_MTE for `target-feature="+mte"` will have no effect.